Library/Homebrew/brew.sh

```diff
@@ -191,6 +191,16 @@ case "$HOMEBREW_COMMAND" in
   --config)    HOMEBREW_COMMAND="config" ;;
 esac
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  export HOMEBREW_GIT_CONFIG_FILE="$HOMEBREW_REPOSITORY/.git/config"
+  HOMEBREW_GIT_CONFIG_DEVELOPERMODE="$(git config --file="$HOMEBREW_GIT_CONFIG_FILE" --get homebrew.developermode)"
```

Comment:
The overhead is tiny (0.004s on my machine) and already exists for analytics checks.





> I can't think of why someone would want HOMEBREW_DEVELOPER unset but have something in the repo's gitconfig saying "I'm a developer".

That's kinda the point of this PR 😉